### PR TITLE
Use "preview" stimulus when exercise is non-interactive

### PR DESCRIPTION
**Pull request by nathanstitt**
The BE now provides the is_interactive and is_video flags, and indicates where in the HTML the preview should be inserted

### CODE REVIEW

File: src/model/exercise.coffee [hunk 1]
```diff
@@ -20,21 +26,22 @@ Exercises =
...
 
   hasInteractive: (exercise) ->
-    !!@getStimulus(exercise).match(/iframe.*(cnx.org|phet.colorado.edu)/)
+    exercise.has_interactive
 
   hasVideo: (exercise) ->
-    !!@getStimulus(exercise).match(/(youtube.com\/(v|embed)\/|khanacademy.org)/)
+    exercise.has_video
+
+  replacePlaceholders: (content) ->
```
Comment (Dantemss) [inline src/model/exercise.coffee:41]: Oh man I thought you were just going to replace this with an image using CSS?

Comment (nathanstitt) [inline src/model/exercise.coffee:41]: yeah, I was planning on doing so, but then realized the placeholder isn't really a SVG, they're components that are rendered.  So then we need to render the strings before/after and then the image in place.  I still like the idea of the div, and have future plans for how we can actually replace them with CSS.

Comment (nathanstitt) [inline src/model/exercise.coffee:41]: the `has_video` and `has_interactive` is still a big win.  at least we'll always get that right for the badges at the top and won't render a honking big sim by mistake

**Comment by philschatz** [comment]
Can you link to the BE PR that adds these fields?

**Comment by Dantemss** [comment]
https://github.com/openstax/tutor-server/pull/1113

File: src/model/exercise.coffee (hunk 1 above)
Comment (philschatz) [inline src/model/exercise.coffee:41]: agree w/ @Dantemss ; pulling out via regexp seems pretty brittle; why weren't you able to add it as a SVG background image in CSS? or replace it in a `componentDidMount`?

**Pull request merged**